nbconvert/exporters/pdf.py

```diff
@@ -154,41 +157,78 @@ def log_error(command, out):
+            # spaces or unicode characters in paths...
+            true_output_dir = resources.get('output_files_dir', cwd)
+            true_output_path = resources.get('metadata', {}).get('path', cwd)
+            resources['output_files_dir'] = os.path.join(td, 'outputs')
+            resources['metadata'] = resources.get('metadata', {})
+            resources['metadata']['path'] = resources['output_files_dir']
+            # We also need to trick the writer to match the temp output dir
+            self.writer.relpath = resources['output_files_dir']
+            self.writer.build_directory = resources['output_files_dir']
+            # Ensure the directory exists ahead of time
+            print(resources['output_files_dir'])
+            self.writer._makedir(resources['output_files_dir'])
+
+            try:
+                yield
+            finally:
+                # Copy any new files back
+                for f in os.listdir(resources['output_files_dir']):
+                    shutil.move(os.path.join(resources['output_files_dir'], f), true_output_dir)
+                resources['output_files_dir'] = true_output_dir
+                resources['metadata']['path'] = true_output_path
+                # Reset the writer
+                self.writer.relpath = true_output_path
+                self.writer.build_directory = true_output_path
+
     def from_notebook_node(self, nb, resources=None, **kw):
-        latex, resources = super(PDFExporter, self).from_notebook_node(
-            nb, resources=resources, **kw
-        )
-        # set texinputs directory, so that local files will be found
-        if resources and resources.get('metadata', {}).get('path'):
-            self.texinputs = resources['metadata']['path']
-        else:
-            self.texinputs = getcwd()
-        
-        self._captured_outputs = []
-        with TemporaryWorkingDirectory():
+        if not resources:
+            resources = {}
+        with self._fake_all_output_paths(resources):
+            latex, resources = super(PDFExporter, self).from_notebook_node(
+                nb, resources=resources, **kw
+            )
+            # set texinputs directory, so that local files will be found
+            if resources and resources.get('metadata', {}).get('path'):
+                self.texinputs = resources['metadata']['path']
+            else:
+                self.texinputs = getcwd()
+
+            self._captured_outputs = []
             notebook_name = 'notebook'
             tex_file = self.writer.write(latex, resources, notebook_name=notebook_name)
             self.log.info("Building PDF")
             rc = self.run_latex(tex_file)
             if rc:
-                rc = self.run_bib(tex_file)
-            if rc:
+                # This can fail for non-conversion blocking reasons
+                self.run_bib(tex_file)
                 rc = self.run_latex(tex_file)
```

Comment:
This second latex should only be run if the bibtex command is successful. Otherwise, the conversion will take almost twice as long as necessary. Maybe nest an ```if``` statement.

Reply:
Good catch, will change that.